sebs/gcp/queue.py

```diff
@@ -0,0 +1,111 @@
+from googleapiclient.discovery import build
+
+from sebs.cache import Cache
+from sebs.faas.config import Resources
+from sebs.faas.queue import Queue, QueueType
+
+from google.api_core import retry
+from google.api_core.exceptions import AlreadyExists
+from google.cloud import pubsub_v1
+
+import os
+
+
+class GCPQueue(Queue):
+    @staticmethod
+    def typename() -> str:
+        return "GCP.Queue"
+
+    @staticmethod
+    def deployment_name():
+        return "gcp"
+
+    @property
+    def topic_name(self):
+        return self._topic_name
+
+    @property
+    def subscription_name(self):
+        return self._subscription_name
+
+    @property
+    def subscription_client(self):
+        return self._subscription_client
+
+    def __init__(
+        self,
+        benchmark: str,
+        queue_type: QueueType,
+        cache_client: Cache,
+        resources: Resources,
+        region: str
+    ):
+        super().__init__(benchmark, queue_type, region, cache_client, resources)
+        self.client = pubsub_v1.PublisherClient()
+        self._subscription_client = pubsub_v1.SubscriberClient()
+
+        self._topic_name = 'projects/{project_id}/topics/{topic}'.format(
+            project_id=os.getenv('GOOGLE_CLOUD_PROJECT'),
```

Comment:
This variable should be stored already in `config.resources` -> no need to retrieve it again.

Reply:
I can't see another mention of `GOOGLE_CLOUD_PROJECT` in the code - I believe it may have only been added in `config.resources` by your NoSQL PR?